archive/imo/imo2008_q4.lean

```diff
@@ -101,7 +101,12 @@ begin
   cases h₃ with hab₁ hab₂,
 
   -- f(ab) = ab → b^4 = 1 → b = 1 → f(b) = b → false
-  { rw hab₁ at H₂, field_simp at H₂,
+  { rw hab₁ at H₂,
+    have H₃ : (a ^ 2 + (1 / b) ^ 2) / (2 * (a * b)) = (a ^ 2 + b ^ 2) / (2 * (a * b)) ↔
+              1 / b ^ 2 = b ^ 2 ∨ 2 * (a * b) = 0,
+    { field_simp,
+      simp [h2ab_ne_0], },
+    rw H₃ at H₂,
```

Comment:
```suggestion
  { have H₃ : (a ^ 2 + (1 / b) ^ 2) / (2 * (a * b)) = (a ^ 2 + b ^ 2) / (2 * (a * b)) ↔
              1 / b ^ 2 = b ^ 2 ∨ 2 * (a * b) = 0,
    { field_simp, simp [h2ab_ne_0], },
    rw [hab₁, H₃] at H₂,
```

Reply:
Can you merge the `field_simp, simp` into `field_simp [h2ab_ne_0]`?

Reply:
Ah it does work, great!